### ISSUE TITLE
invalid key name "autoscaling.knative.dev/scale-down-delay"

### ISSUE THREAD

**Issue by kgibm** [OpenLiberty/open-liberty-operator#474]
### Is your feature request related to a problem?

I have an issue when trying to set `autoscaling.knative.dev/scale-down-delay`:

```
$ oc describe webspherelibertyapplication surveyinputservice
[...]
  Warning  ProcessingError  1s (x4 over 16s)   websphere-liberty-operator  admission webhook "validation.webhook.serving.knative.dev" denied the request: validation failed: invalid key name "autoscaling.knative.dev/scale-down-delay": metadata.annotations
autoscaling annotations must be put under "spec.template.metadata.annotations" to work
```

Based on:

```
metadata:
  annotations:
    autoscaling.knative.dev/scale-down-delay: "2700s"
```

### Describe the solution you'd like

Some way to specify KNative annotations.

**Comment by leochr** [comment]
**Potential workarounds:**
- set the annotation directly on Knative Service once Liberty Operator has created it (under `spec.template.metadata.annotations`)
- set the relevant config using Global ConfigMap config-autoscaler in knative-serving namespace
- set the relevant config using KnativeServing

more info: https://knative.dev/docs/serving/autoscaling/scale-bounds/#scale-down-delay

------

**Cause:**
This appears to be due to a new validation check being done Knative/Serverless.

The annotations added to OpenLibertyApplication’s metadata are propagated down to all the resources that are created by the Liberty Operator, including the Knative Service. They are also set at `spec.template.metadata.annotations` in Knative Service. However, due to the validation error, the value is not actually set on the Knative Service.

------

**Fix:**
We need to do some filtering on the annotations. Exclude the ones starting with `autoscaling.knative.dev/` when setting  `metadata.annotations` of Knative Service. They should still be set at `spec.template.metadata.annotations` in Knative Service

**Comment by idlewis** [comment]
This needs fixing in RCO, OLO,  and WLO:

- [ ] RCO
- [ ] OLO
- [ ] WLO

**Comment by idlewis** [comment]
All fixes delivered for this issue, should be available in the upcoming 1.3.0 operator release

**Comment by idlewis** [comment]
I'm closing this issue as 1.3.0 has now been released.
